tests/lib/rules/no-unnecessary-act.test.ts

```diff
@@ -0,0 +1,782 @@
+import { createRuleTester } from '../test-utils';
+import rule, { RULE_NAME } from '../../../lib/rules/no-unnecessary-act';
+
+const ruleTester = createRuleTester();
+
+/**
+ * - AGR stands for Aggressive Reporting
+ * - RTL stands for React Testing Library (@testing-library/react)
+ * - RTU stands for React Test Utils (react-dom/test-utils)
+ */
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    {
+      code: `// case: RTL act wrapping non-RTL calls
+      import { act } from '@testing-library/react'
+
+      test('valid case', async () => {
+        act(() => {
+          stuffThatDoesNotUseRTL();
+        });
+
+        await act(async () => {
+          stuffThatDoesNotUseRTL();
+        });
+
+        act(function() {
+          stuffThatDoesNotUseRTL();
+          const a = foo();
+        });
+
+        act(function() {
+          return stuffThatDoesNotUseRTL();
+        });
+
+        act(() => stuffThatDoesNotUseRTL());
+      });
+      `,
+    },
+    {
+      code: `// case: RTU act wrapping non-RTL
+      import { act } from 'react-dom/test-utils'
+
+      test('valid case', async () => {
+        act(() => {
+          stuffThatDoesNotUseRTL();
+        });
+
+        await act(async () => {
+          stuffThatDoesNotUseRTL();
+        });
+
+        act(function() {
+          stuffThatDoesNotUseRTL();
+        });
+
+        act(function() {
+          return stuffThatDoesNotUseRTL();
+        });
+
+        act(() => stuffThatDoesNotUseRTL());
+      });
+      `,
+    },
+    {
+      settings: {
+        'testing-library/utils-module': 'test-utils',
+      },
+      code: `// case: RTL act wrapping non-RTL - AGR disabled
+      import { act } from '@testing-library/react'
+      import { waitFor } from 'somewhere-else'
+
+      test('valid case', async () => {
+        act(() => {
+          waitFor();
+        });
+
+        await act(async () => {
+          waitFor();
+        });
+
+        act(function() {
+          waitFor();
+        });
+
+        act(function() {
+          return waitFor();
+        });
+
+        act(() => waitFor());
+      });
+      `,
+    },
+    {
+      code: `// case: RTL act wrapping both RTL and non-RTL calls
+      import { act, render, waitFor } from '@testing-library/react'
+
+      test('valid case', async () => {
+        act(() => {
+          render(element);
+          stuffThatDoesNotUseRTL();
+        });
+
+        await act(async () => {
+          waitFor();
+          stuffThatDoesNotUseRTL();
```

Comment:
`await stuffThatDoesNotUseRTL();` This would make the test fail. Is it intended?

Reply:
On top of that, this test case doesn't pass, although I think that's valid:

```
        await act(async () => {
          await stuffThatDoesNotUseRTL();
        });
```

Reply:
Mmm not really. Fail in terms of reporting an unexpected error or throwing an exception?

Reply:
> Mmm not really. Fail in terms of reporting an unexpected error or throwing an exception?

Reports unexpected error (`no-unnecessary-act`), if you're referring to my 1st comment

Reply:
I got it. Basically, I ignoring`AwaitExpression` and `AssigmentExpression`. Potentially, [all different expressions listed here](https://eslint-utils.mysticatea.dev/api/ast-utils.html#hassideeffect) would be checked here, but I think the current list is enough.